plugins/jvm-abi-gen/src/org/jetbrains/kotlin/jvm/abi/JvmAbiClassBuilderInterceptor.kt

```diff
@@ -110,6 +118,32 @@ class JvmAbiClassBuilderInterceptor(
             delegate.visitEnclosingMethod(owner, name, desc)
         }
 
+        override fun newField(
+            declaration: IrField?, access: Int, name: String, desc: String, signature: String?, value: Any?
+        ): FieldVisitor {
+            if (keepClassAsIs || removeClassFromAbi) {
+                // We don't care about fields when we remove or keep this class completely.
+                return delegate.newField(declaration, access, name, desc, signature, value)
+            }
+
+            val visibility = declaration?.visibility ?: DescriptorVisibilities.DEFAULT_VISIBILITY
+
+            if (DescriptorVisibilities.isPrivate(visibility)) {
+                //remove all private fields
+                return delegate.newField(declaration, access, name, desc, signature, value)
+            }
+
+            if (removeInternalDeclarations && visibility == DescriptorVisibilities.INTERNAL) {
+                //remove all internal fields
+                return delegate.newField(declaration, access, name, desc, signature, value)
+            }
+
+            //keep otherwise
```

Comment:
```
// Please maintain the comment style everywhere.
```

Reply:
It's like a good old friend to me at this point :D